sequencer/src/bin/test-node.rs

```diff
@@ -0,0 +1,312 @@
+use std::{io, time::Duration};
+
+use async_compatibility_layer::logging::{setup_backtrace, setup_logging};
+use async_std::task::spawn;
+use clap::Parser;
+use es_version::SEQUENCER_VERSION;
+use ethers::types::Address;
+use futures::FutureExt;
+use sequencer::{
+    api::options,
+    hotshot_commitment::{run_hotshot_commitment_task, CommitmentTaskOptions},
+    persistence,
+    test_helpers::TestNetwork,
+    testing::TestConfig,
+};
+use sequencer_utils::{
+    deployer::{deploy, Contract, Contracts},
+    AnvilOptions,
+};
+use tide_disco::{error::ServerError, Api};
+use url::Url;
+use vbs::version::StaticVersionType;
+
+#[derive(Clone, Debug, Parser)]
+struct Options {
+    /// A JSON-RPC endpoint for the L1 to deploy to.
+    #[clap(short, long, env = "ESPRESSO_SEQUENCER_L1_PROVIDER")]
+    rpc_url: Option<Url>,
+    /// Mnemonic for an L1 wallet.
+    ///
+    /// This wallet is used to deploy the contracts, so the account indicated by ACCOUNT_INDEX must
+    /// be funded with with ETH.
+    #[clap(
+        long,
+        name = "MNEMONIC",
+        env = "ESPRESSO_SEQUENCER_ETH_MNEMONIC",
+        default_value = "test test test test test test test test test test test junk"
+    )]
+    mnemonic: String,
+    /// Account index in the L1 wallet generated by MNEMONIC to use when deploying the contracts.
+    #[clap(
+        long,
+        name = "ACCOUNT_INDEX",
+        env = "ESPRESSO_DEPLOYER_ACCOUNT_INDEX",
+        default_value = "0"
+    )]
+    account_index: u32,
+    /// Port to run the builder server on.
+    #[clap(short, long, env = "ESPRESSO_BUILDER_SERVER_PORT")]
```

Comment:
I think we can put some defaults here for convenience. Like 3 consecutive ports that aren't by default used by any other tools we use. For example, maybe

API: 8770
Builder: 8771
Commit: 8772

Reply:
[b042b8e1](https://github.com/EspressoSystems/espresso-sequencer/pull/1353/commits/b042b8e1af4adf715117ab8858d589d951fd5018)